src/main/java/Duke.java

```diff
@@ -255,13 +266,85 @@ private static void addEvent(String userInput) {
         }
 
         // checks if date is correctly formatted and valid
-        if (!placementCorrect) {
+        if (placementCorrect) {
+            // for start time
+            boolean isStartDateWithTime =
+                    startTimeUnformatted.matches("([0-9]{2})-([0-9]{2})-([0-9]{4}) ([0-9]{2}):([0-9]{2})");
+            boolean isStartDateWithoutTime = startTimeUnformatted.matches("([0-9]{2})-([0-9]{2})-([0-9]{4})");
+            // for end time
+            boolean isEndDateWithTime =
+                    endTimeUnformatted.matches("([0-9]{2})-([0-9]{2})-([0-9]{4}) ([0-9]{2}):([0-9]{2})");
+            boolean isEndDateWithoutTime = endTimeUnformatted.matches("([0-9]{2})-([0-9]{2})-([0-9]{4})");
+
+            if (isStartDateWithTime) {
+                try {
+                    startTime = dateWithTime.parse(startTimeUnformatted);
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
+            } else if(isStartDateWithoutTime){
+                try {
+                    startTime = dateWithoutTime.parse(startTimeUnformatted);
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
+            } else {
+                System.out.println("Check start date formatting!" + NEWLINE + NEWLINE + EVENT_COMMAND_DESCRIPTION);
+                return;
+            }
+
+            if (isEndDateWithTime) {
+                try {
+                    endTime = dateWithTime.parse(endTimeUnformatted);
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                    return;
+                }
+            } else if(isEndDateWithoutTime){
+                try {
+                    endTime = dateWithoutTime.parse(endTimeUnformatted);
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                    return;
+                }
+            } else {
+                System.out.println("Check end date formatting!" + NEWLINE + NEWLINE + EVENT_COMMAND_DESCRIPTION);
+                return;
+            }
+
+            Calendar cal = Calendar.getInstance();
+            cal.setLenient(false);
+            cal.setTime(startTime);
+            try {
+                cal.getTime();
+            }
+            catch (Exception e) {
+                System.out.println("Invalid start date");
+                return;
+            }
+
+            cal.setLenient(false);
+            cal.setTime(endTime);
+            try {
+                cal.getTime();
+            }
+            catch (Exception e) {
+                System.out.println("Invalid end date");
+                return;
+            }
```

Comment:
Formatting of try-catch statements are inconsistent.
Some are formatted as such:
`try {`
`----endTime = dateWithoutTime.parse(endTimeUnformatted);`
`} catch (ParseException e) {`
`----e.printStackTrace();`
`----return;`
`}`

And the rest are formatted as the highlighted lines.